src/query/string_opfunc.c

```diff
@@ -4481,6 +4481,466 @@ db_string_rlike (const DB_VALUE * src, const DB_VALUE * pattern, const DB_VALUE
+	er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, error_status, 0);
+	goto exit;
+      }
+
+    // *INDENT-OFF*
+    /* match_type argument check */
+    std::regex_constants::syntax_option_type reg_flags = std::regex_constants::ECMAScript | std::regex_constants::icase;
+    if (match_type)
+      {
+	std::string match_type_str (db_get_string (match_type), db_get_string_size (match_type));
+	error_status = cubregex::parse_match_type (reg_flags, match_type_str);
+	if (error_status != NO_ERROR)
+	  {
+	    er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, error_status, 0);
+	    goto exit;
+	  }
+      }
+    // *INDENT-ON*
+
+    /* check pattern string */
+    if (db_get_string_size (pattern) == 0 || position_value >= db_get_string_size (src))
+      {
+	goto exit;
+      }
+
+    LANG_COLLATION *collation = lang_get_collation (coll_id);
+    assert (collation != NULL);
+
+    // *INDENT-OFF*
+    /* compile pattern if needed */
+    std::string pattern_string (db_get_string (pattern), db_get_string_size (pattern));
+    if (cubregex::check_should_recompile (rx_compiled_regex, rx_compiled_pattern, pattern_string, reg_flags) == true)
+      {
+	cubregex::clear (rx_compiled_regex, rx_compiled_pattern);
+	int pattern_length = pattern_string.size ();
+	rx_compiled_pattern = (char *) db_private_alloc (NULL, pattern_length + 1);
+	if (rx_compiled_pattern == NULL)
+	  {
+	    /* out of memory */
+	    error_status = ER_OUT_OF_VIRTUAL_MEMORY;
+	    er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, error_status, 0);
+	    goto exit;
+	  }
+	memcpy (rx_compiled_pattern, pattern_string.c_str (), pattern_length);
+	rx_compiled_pattern[pattern_length] = '\0';
+
+	error_status = cubregex::compile (rx_compiled_regex, rx_compiled_pattern, reg_flags, collation);
+	if (error_status != NO_ERROR)
+	  {
+	error_status = (error_status == ER_QSTR_BAD_SRC_CODESET) ? NO_ERROR : error_status;
+  goto exit;
+	  }
+      }
+
+    /* perform regular expression according to the position and occurence value */
+    int result_value = 0;
+    std::string src_string (db_get_string (src), db_get_string_size (src));
+    error_status = cubregex::instr (result_value, *rx_compiled_regex, src_string, position_value,
+				      occurrence_value, return_opt_value, collation->codeset);
+    if (error_status != NO_ERROR)
+      {
+	/* regex execution error */
+	error_status = (error_status == ER_QSTR_BAD_SRC_CODESET) ? NO_ERROR : error_status;
+  goto exit;
+      }
+    // *INDENT-ON*
+
+    /* make result */
+    db_make_int (result, result_value);
+    goto exit;
+  }
+
+exit:
+  if (error_status != NO_ERROR)
+    {
+      db_make_null (result);
+      // *INDENT-OFF*
+      cubregex::clear (rx_compiled_regex, rx_compiled_pattern);
+      // *INDENT-ON*
+      if (prm_get_bool_value (PRM_ID_RETURN_NULL_ON_FUNCTION_ERRORS))
+	{
+	  /* we must not return an error code */
+	  er_clear ();
+	  error_status = NO_ERROR;
+	}
+    }
+  // *INDENT-ON*
+
+  if (comp_regex == NULL || comp_pattern == NULL)
```

Comment:
Since comp_regex and comp_pattern can be assigned separately, I think these pointer variable should be checked one by one.
```
if (comp_regex != NULL)
    {
      *comp_regex = rx_compiled_regex;
    }
  if (comp_pattern != NULL)
    {
      *comp_pattern = rx_compiled_pattern;
    }
```

Reply:
I fixed it